InterviewFlashCards/IFCFlashCard.m

```diff
@@ -86,9 +86,13 @@ - (void)prepareFlashCardWithCompletion:(void (^)())completion{
     } else {
         self.answerImagesLoaded = YES;
         if (self.questionImagesLoaded) {
-            dispatch_async(dispatch_get_main_queue(), ^{
+            if (NSThread.isMainThread) {
                 completion();
-            });
+            } else {
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    completion();
+                });
+            }
```

Comment:
We were causing a delayed reaction by unnecessarily dispatching to the main queue even when on the main thread already.